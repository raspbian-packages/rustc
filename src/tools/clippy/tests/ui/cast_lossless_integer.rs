
#[warn(cast_lossless)]
#[allow(no_effect, unnecessary_operation)]
fn main() {
    // Test cast_lossless with casts to integer types
    1i8 as i16;
    1i8 as i32;
    1i8 as i64;
    1u8 as i16;
    1u8 as i32;
    1u8 as i64;
    1u8 as u16;
    1u8 as u32;
    1u8 as u64;
    1i16 as i32;
    1i16 as i64;
    1u16 as i32;
    1u16 as i64;
    1u16 as u32;
    1u16 as u64;
    1i32 as i64;
    1u32 as i64;
    1u32 as u64;
}
