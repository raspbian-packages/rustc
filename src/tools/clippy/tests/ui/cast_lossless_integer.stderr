error: casting i8 to i16 may become silently lossy if types change
 --> $DIR/cast_lossless_integer.rs:6:5
  |
6 |     1i8 as i16;
  |     ^^^^^^^^^^ help: try: `i16::from(1i8)`
  |
  = note: `-D cast-lossless` implied by `-D warnings`

error: casting i8 to i32 may become silently lossy if types change
 --> $DIR/cast_lossless_integer.rs:7:5
  |
7 |     1i8 as i32;
  |     ^^^^^^^^^^ help: try: `i32::from(1i8)`

error: casting i8 to i64 may become silently lossy if types change
 --> $DIR/cast_lossless_integer.rs:8:5
  |
8 |     1i8 as i64;
  |     ^^^^^^^^^^ help: try: `i64::from(1i8)`

error: casting u8 to i16 may become silently lossy if types change
 --> $DIR/cast_lossless_integer.rs:9:5
  |
9 |     1u8 as i16;
  |     ^^^^^^^^^^ help: try: `i16::from(1u8)`

error: casting u8 to i32 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:10:5
   |
10 |     1u8 as i32;
   |     ^^^^^^^^^^ help: try: `i32::from(1u8)`

error: casting u8 to i64 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:11:5
   |
11 |     1u8 as i64;
   |     ^^^^^^^^^^ help: try: `i64::from(1u8)`

error: casting u8 to u16 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:12:5
   |
12 |     1u8 as u16;
   |     ^^^^^^^^^^ help: try: `u16::from(1u8)`

error: casting u8 to u32 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:13:5
   |
13 |     1u8 as u32;
   |     ^^^^^^^^^^ help: try: `u32::from(1u8)`

error: casting u8 to u64 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:14:5
   |
14 |     1u8 as u64;
   |     ^^^^^^^^^^ help: try: `u64::from(1u8)`

error: casting i16 to i32 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:15:5
   |
15 |     1i16 as i32;
   |     ^^^^^^^^^^^ help: try: `i32::from(1i16)`

error: casting i16 to i64 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:16:5
   |
16 |     1i16 as i64;
   |     ^^^^^^^^^^^ help: try: `i64::from(1i16)`

error: casting u16 to i32 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:17:5
   |
17 |     1u16 as i32;
   |     ^^^^^^^^^^^ help: try: `i32::from(1u16)`

error: casting u16 to i64 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:18:5
   |
18 |     1u16 as i64;
   |     ^^^^^^^^^^^ help: try: `i64::from(1u16)`

error: casting u16 to u32 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:19:5
   |
19 |     1u16 as u32;
   |     ^^^^^^^^^^^ help: try: `u32::from(1u16)`

error: casting u16 to u64 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:20:5
   |
20 |     1u16 as u64;
   |     ^^^^^^^^^^^ help: try: `u64::from(1u16)`

error: casting i32 to i64 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:21:5
   |
21 |     1i32 as i64;
   |     ^^^^^^^^^^^ help: try: `i64::from(1i32)`

error: casting u32 to i64 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:22:5
   |
22 |     1u32 as i64;
   |     ^^^^^^^^^^^ help: try: `i64::from(1u32)`

error: casting u32 to u64 may become silently lossy if types change
  --> $DIR/cast_lossless_integer.rs:23:5
   |
23 |     1u32 as u64;
   |     ^^^^^^^^^^^ help: try: `u64::from(1u32)`

error: aborting due to 18 previous errors

